### PR TITLE
add glib/2.64.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **glib/2.64.0**

this is a port of https://github.com/bincrafters/conan-glib but it does not support windows (not possible until https://github.com/conan-io/conan-center-index/pull/1013 is merged)

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
Contrary to what the bot says, the actual error (which can be seen in create_stdout.txt) is the usual
:
```
ninja: Entering directory `/tmp/c3ipr/pr_1756_2_0_8/.conan/data/glib/2.64.0/_/_/build/3a02e4515e73052f165a2fb983aee7bb47a556a7/build_subfolder'
ninja: relocation error: ninja: symbol _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE7reserveEm, version GLIBCXX_3.4.21 not defined in file libstdc++.so.6 with link time reference
```

File: recipes/glib/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+               "with_mount": [True, False]}
+    default_options = {"shared": True,
+                       "fPIC": True,
+                       "with_pcre": True,
+                       "with_elf": True,
+                       "with_mount": True,
+                       "with_selinux": True}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = 'build_subfolder'
+    autotools = None
```
Comment (uilianries) [inline recipes/glib/all/conanfile.py:30]:
```suggestion
```

unused

File: recipes/glib/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            self.options.shared = True
```
Comment (uilianries) [inline recipes/glib/all/conanfile.py:43]:
```suggestion
        if self.settings.os == "Windows" and not self.options.shared:
            raise ConanInvalidConfiguration("glib can not be built as static on Windows.")
```

File: recipes/glib/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    autotools = None
+    short_paths = True
+    generators = "pkg_config"
+    requires = "zlib/1.2.11", "libffi/3.3"
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def configure(self):
```
Comment (uilianries) [inline recipes/glib/all/conanfile.py:39]:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC
```

**Pull request merged**